lib/rspec/matchers/built_in/match.rb

```diff
@@ -5,6 +5,22 @@ module BuiltIn
       # Provides the implementation for `match`.
       # Not intended to be instantiated directly.
       class Match < BaseMatcher
+        # Used to specify the captures we match against
+        # Accepts variadic non-named captures
+        # @return [self]
+        def with_captures(*captures)
+          @expected_captures = captures
+          extend WithCaptures
+        end
```

Comment:
I have two problems with this, firstly dynamic `extend`? Yuk. Both from a stylistic pov and a performance pov. Second, this allows `with_captures` on strings, which is non ideal. I prefer @samphippen's dual class approach for this.